src/pyscaffold/shell.py

```diff
@@ -191,7 +214,39 @@ def get_editor(**kwargs):
 
 def edit(file: PathLike, *args, **kwargs) -> Path:
     """Open a text editor and returns back a :obj:`Path` to file, after user editing."""
-    editor = ShellCommand(get_editor())
+    editor = ShellCommand(_quote(get_editor()))
+    # ^  since we quote here we cannot pass flags in EDITOR environment variable
     editor(file, *args, **{"stdout": None, "stderr": None, **kwargs})
     # ^  stdout/stderr=None => required for a terminal editor to open properly
     return Path(file)
+
+
+def _has_whitespace(executable_path: str) -> bool:  # pragma: no cover
+    # This helper function is only supposed to be used on Windows
+    # and in our current CI setup, even if it runs, the coverage data will not be
+    # submitted
+    parts = executable_path.split()
+    # by default split will consider all kinds of whitespace
+    return len(parts) > 1
+
+
+def _quote(executable_path: str) -> str:
+    """Prevent whitespaces for breaking executable paths when called from
+    shell.
+
+    Please note this is not intended to quote everything, only executable paths.
+    This assumption is important because we don't have to concern about special
+    characters that are disallowed in file names. For example, on Windows
+    ``"`` chars are reserved characters and not supposed to appear if file
+    paths (https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file).
+    """
+    if os.name == "posix":
+        return shlex.quote(executable_path)
+        # shlex is supposed to not quote if not necessary
+    elif os.name == "nt" and _has_whitespace(executable_path):  # pragma: no cover
+        # in our current CI setup, Windows-specific code might run during tests
+        # but the coverage data is not submitted
+        return f'"{executable_path}"'
```

Comment:
This quote method should be fine, since quote symbols cannot be used in file paths on Windows... However quoting is always trick...

An alternative could be using Windows short path, as described in https://stackoverflow.com/questions/23598289/how-to-get-windows-short-file-name-in-python.

It would be a more convoluted way, but I suppose more robust.

Reply:
Couldn't we just always quote the executable? Doesn't for instance `"ls"` work in unix environments?

Reply:
I am always insecure when it comes to quoting... This code is always invoking `shlex.quote` for POSIX, and I trust `shlex` knows what is best.

Unfortunately `shlex.quote` does not work well for the Windows.

I think I prefer not to quote if I don't have to, because there might be edge cases... (and the topic is complex: https://docs.microsoft.com/en-gb/archive/blogs/twistylittlepassagesallalike/everyone-quotes-command-line-arguments-the-wrong-way)